### PR TITLE
Added Zip to the list of linq functions

### PR DESCRIPTION
**Pull request by psarathi**

### CODE REVIEW

File: linq.go [hunk 1]
```diff
@@ -1218,6 +1219,42 @@ func (q Query) Average() (avg float64, err error) {
 	return
 }
 
+// Zip returns a query with the result of the "zipping" function
+// provided
+//
+// Example:
+// convertRoman:= func(n, m T) (T, error){
```
Comment (ahmetalpbalkan) [inline linq.go]: Can you please indent the examples like the functions above and have proper gofmt'ing (like space before :=)

File: linq.go [hunk 2]
```diff
@@ -1218,6 +1219,42 @@ func (q Query) Average() (avg float64, err error) {
...
+	if f == nil {
+		r.err = ErrNilFunc
+		return
+	}
+	q2 := From(second)
+	if q2.err != nil {
+		r.err = q2.err
+		return r
+	}
+	maxIterationCount := int (math.Min(float64(len(q.values)), float64(len(q2.values))))
```
Comment (ahmetalpbalkan) [inline linq.go]: This line is not properly gofmt'ed. Please run gofmt on these files.

Comment (ahmetalpbalkan) [inline linq.go]: Also you can use `max=b; if a>b{max=a}`. No need to convert to float and use math.floor here

**Comment by ahmetalpbalkan** [comment]
@psarathi thanks for your first contribution! This generally looks nice to me. If you could address the comments above, I can merge.  (just create more commits and push to your branch, the pull request will automatically get updated.)

**Comment by psarathi** [comment]
@ahmetalpbalkan thanks for your suggestions. I'll make the changes and push them soon.

**Comment by ahmetalpbalkan** [comment]
Sorry I totally forgot about this. LGTM. Thanks @psarathi.

**Comment by psarathi** [comment]
Thanks @ahmetalpbalkan

**Pull request merged**